src/pal/src/loader/module.cpp

```diff
@@ -1628,6 +1636,12 @@ static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
     LockModuleList();
 
     dl_handle = LOADLoadLibraryDirect(shortAsciiName);
+#ifdef __APPLE__
+    if (!dl_handle && isLibcRequested)
+    {
+        dl_handle = LOADLoadLibraryDirect("/usr/lib/libc.dylib");
```

Comment:
I would also prefer using the full path only on OSX. I have found the full path mentioned in **Mac OS X Internals: A Systems Approach** as the path of the libc. This book was printed in 2006, so it seems pretty stable:
https://books.google.cz/books?id=K8vUkpOXhN4C&pg=PA74&lpg=PA74&dq=libc.dylib+location&source=bl&ots=OLnnXYSwXy&sig=D1CMLFhuHQdMcnEw9J7Kq6KoVA8&hl=cs&sa=X&ved=0ahUKEwjL5o2vwP3NAhWKCMAKHRS0BpcQ6AEIXzAI#v=onepage&q=libc.dylib%20location&f=false